src/Xamarin.Android.Build.Tasks/Resources/MultiDexLoader.Debug.java

```diff
@@ -0,0 +1,233 @@
+		return dir + "/";
+	}
+
+	private String getExternalResourceFile () {
+		String base = mIncrementalDeploymentDir;
+		String resourceFile = base + ".__override__/packaged/packaged_resources";
+		if (!(new File (resourceFile).isFile ())) {
+			resourceFile = base + ".__override__/packaged_resources";
+			if (!(new File (resourceFile).isFile ())) {
+				resourceFile = base + ".__override__/resources";
+				if (!(new File (resourceFile).isDirectory ())) {
+					Log.v ("MultiDexLoader", "Cannot find external resources, not patching them in");
+					return null;
+				}
+			}
+		}
+
+		Log.v ("MultiDexLoader", "Found external resources at " + resourceFile);
+		return resourceFile;
+	}
+
+	private List<String> getDexList (String packageName)
+	{
+		List<String> result = new ArrayList<String> ();
+		String dexDirectory = mIncrementalDeploymentDir + ".__override__/dexes";
+		File[] dexes = new File (dexDirectory).listFiles ();
+		// It is not illegal state when it was launched to start Seppuku
+		if (dexes == null) {
+			Log.v("MultiDexLoader", "No dexes!");
+			return null;
+		} else {
+			for (File dex : dexes) {
+				if (dex.getName ().endsWith (".dex")) {
+					Log.v("MultiDexLoader", "Adding dex " + dex.getPath ());
+					result.add (dex.getPath ());
+				}
+			}
+		}
+
+		return result;
+	}
+	
+	// ---
+	@Override
+	public android.database.Cursor query (android.net.Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder)
+	{
+		throw new RuntimeException ("This operation is not supported.");
+	}
+
+	@Override
+	public String getType (android.net.Uri uri)
+	{
+		throw new RuntimeException ("This operation is not supported.");
+	}
+
+	@Override
+	public android.net.Uri insert (android.net.Uri uri, android.content.ContentValues initialValues)
+	{
+		throw new RuntimeException ("This operation is not supported.");
+	}
+
+	@Override
+	public int delete (android.net.Uri uri, String where, String[] whereArgs)
+	{
+		throw new RuntimeException ("This operation is not supported.");
+	}
+
+	@Override
+	public int update (android.net.Uri uri, android.content.ContentValues values, String where, String[] whereArgs)
+	{
+		throw new RuntimeException ("This operation is not supported.");
+	} 
+	public static List<Activity> getActivities(Context context, boolean foregroundOnly)
+	{
+		List<Activity> list = new ArrayList<Activity>();
+		try {
+			Class activityThreadClass = Class.forName("android.app.ActivityThread");
+			Object activityThread = MonkeyPatcher.getActivityThread(context, activityThreadClass);
+			Field activitiesField = activityThreadClass.getDeclaredField("mActivities");
+			activitiesField.setAccessible(true);
+			Collection c;
+			Object collection = activitiesField.get(activityThread);
+			if (collection instanceof HashMap) {
+				// Older platforms
+				Map activities = (HashMap) collection;
+				c = activities.values();
+			} else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT &&
```

Comment:
This will probably fail when building on e.g. API-10, as `Build.VERSION_CODES.KITKAT` won't exist.